modules/platforms/cpp/thin-client/include/ignite/thin/compute/compute_client.h

```diff
@@ -0,0 +1,201 @@
+                 */
+                ComputeClient(const SP_Void& impl) :
+                    impl(impl),
+                    flags(ComputeClientFlags::NONE),
+                    timeout(0)
+                {
+                    // No-op.
+                }
+
+                /**
+                 * Destructor.
+                 */
+                ~ComputeClient()
+                {
+                    // No-op.
+                }
+
+                /**
+                 * Executes given Java task by class name.
+                 *
+                 * @param taskName Java task name.
+                 * @param taskArg Argument of task execution of type A.
+                 * @return Task result of type @c R.
+                 *
+                 * @tparam R Type of task result.
+                 * @tparam A Type of task argument.
+                 */
+                template<typename R, typename A>
+                R ExecuteJavaTask(const std::string& taskName, const A& taskArg)
+                {
+                    R result;
+
+                    impl::thin::WritableImpl<A> wrArg(taskArg);
+                    impl::thin::ReadableImpl<R> rdResult(result);
+
+                    InternalExecuteJavaTask(taskName, wrArg, rdResult);
+
+                    return result;
+                }
+
+                /**
+                 * Executes given Java task by class name.
+                 *
+                 * @param taskName Java task name.
+                 * @return Task result of type @c R.
+                 *
+                 * @tparam R Type of task result.
+                 */
+                template<typename R>
+                R ExecuteJavaTask(const std::string& taskName)
+                {
+                    R result;
+
+                    impl::thin::WritableImpl<int*> wrArg(0);
+                    impl::thin::ReadableImpl<R> rdResult(result);
+
+                    InternalExecuteJavaTask(taskName, wrArg, rdResult);
+
+                    return result;
+                }
+
+                /**
+                 * Returns a new instance of ComputeClient with a timeout for all task executions.
+                 *
+                 * @param timeoutMs Timeout in milliseconds.
+                 * @return New ComputeClient instance with timeout.
+                 */
+                ComputeClient WithTimeout(int64_t timeoutMs)
+                {
+                    return ComputeClient(impl, flags, timeoutMs);
+                }
+
+                /**
+                 * Returns a new instance of ComputeClient with disabled failover.
+                 * When failover is disabled, compute jobs won't be retried in case of node crashes.
+                 *
+                 * @return New Compute instance with disabled failover.
+                 */
+                ComputeClient WithNoFailover()
+                {
+                    return ComputeClient(impl, flags | ComputeClientFlags::NO_FAILOVER, timeout);
```

Comment:
When current instance has the flag, we can return it as is.

Reply:
That's a good point. Will fix this for other flag also.